web_src/js/features/tribute.js

```diff
@@ -0,0 +1,61 @@
+import {emojiKeys, emojiMap, giteaImage} from './emoji.js';
+
+export const issuesTribute = window.config.Tribute ? new Tribute({
+  values: window.config.tributeValues,
+  noMatchTemplate() { return null },
+  menuItemTemplate(item) {
+    const div = $('<div/>');
+    div.append($('<img/>', {src: item.original.avatar}));
+    div.append($('<span/>', {class: 'name'}).text(item.original.name));
+    if (item.original.fullname && item.original.fullname !== '') {
+      div.append($('<span/>', {class: 'fullname'}).text(item.original.fullname));
+    }
+    return div.html();
+  }
+}) : null;
+
+export const emojiTribute = window.config.Tribute ? new Tribute({
+  collection: [{
+    trigger: ':',
+    requireLeadingSpace: true,
+    values(query, cb) {
+      const matches = [];
+      for (const name of emojiKeys) {
+        if (name.includes(query)) {
+          matches.push(name);
+          if (matches.length > 5) break;
+        }
+      }
+      cb(matches);
+    },
+    lookup(item) {
+      return item;
+    },
+    selectTemplate(item) {
+      if (typeof item === 'undefined') return null;
+      return `:${item.original}:`;
+    },
+    menuItemTemplate(item) {
+      return `<div class="tribute-item"><span class="emoji">${item.original === 'gitea' ? giteaImage : emojiMap[item.original]}</span>${item.original}</div>`;
```

Comment:
FYI I plan to refactor this into a shared function that returns emoji HTML from a keyword.

Reply:
And it's done.